vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/utils/DateUtils.kt

```diff
@@ -15,27 +16,67 @@ import io.mosip.vercred.vcverifier.exception.ValidationException
 import org.json.JSONObject
 import java.text.SimpleDateFormat
 import java.util.Calendar
+import java.util.Date
 import java.util.Locale
 import java.util.TimeZone
 
-class DateUtils {
+object DateUtils {
+
+    private val dateFormats = listOf(
+        ("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"),
+        ("yyyy-MM-dd'T'HH:mm:ss'Z'")
+    )
+
+    private const val UTC = "UTC"
+
     fun isValidDate(dateValue: String): Boolean {
         return DATE_REGEX.matches(dateValue)
     }
 
     fun isDatePassedCurrentDate(inputDateString: String): Boolean {
         return try {
-            val format = SimpleDateFormat(COMMON_DATE_FORMAT, Locale.getDefault()).apply {
-                timeZone = TimeZone.getTimeZone(UTC)
+            val inputDate: Date? = parseDate(inputDateString)
```

Comment:
It has impacted dateformat used for ldp vc.

Reply:
👍 will check if this dateformat can be extracted out so that it can be reused w/o breaking

Reply:
Checked in with @balachandar-tw reg this,
1. w3 org [issuanceDate / expiration](https://www.w3.org/TR/xmlschema11-2/#dateTime) suggests the date value to be of regex [[XMLSCHEMA11-2](https://www.w3.org/TR/xmlschema11-2/#dateTime)]

`The [dateTimeLexicalRep](https://www.w3.org/TR/xmlschema11-2/#nt-dateTimeRep) production is equivalent to this regular expression once whitespace is removed.
-?([1-9][0-9]{3,}|0[0-9]{3})
-(0[1-9]|1[0-2])
-(0[1-9]|[12][0-9]|3[01])
T(([01][0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9](\.[0-9]+)?|(24:00:00(\.0+)?))
(Z|(\+|-)((0[0-9]|1[0-3]):[0-5][0-9]|14:00))?`

Here the seconds fragment has milliSeconds as optional, so DateUtils, dateFormats involving pattern w/o millseconds would not be breaking the standard for ldp_vc

2. isValidDate check is performed for ldp_vc as per the XMLSCHEME11-2 standard, so if any ldp_vc is not following the standard it would be caught up in that isValidDate check making this DateUtils. isDatePassedCurrentDate to be independent of date validation as per VC format standards